.github/scripts/label_promoted_commits.py

```diff
@@ -35,7 +35,24 @@ def add_comment_and_close_pr(pr_number, repository, comment):
         pr.edit(state="closed")
 
 
+def switch_label_after_backport(repo, pr_number, label_to_add, label_to_remove=None, sha=None, backport_pr_number=None):
+    pr = repo.get_pull(pr_number)
+    current_labels = pr.get_labels()
+    current_label_names = [label.name for label in current_labels]
+    if label_to_add not in current_label_names:
+        pr.add_to_labels(label_to_add)
+    if label_to_remove and sha and backport_pr_number:
+        if label_to_remove in current_label_names:
+            pr.remove_from_labels(label_to_remove)
+        comment = f'Closed via {sha}'
+        add_comment_and_close_pr(backport_pr_number, repo, comment)
+
+
 def main():
+    # This script will be trigger following a push event (either to master or branch-x.y), and will add the proper label based on the following roles:
+    # - When ref branch is `master`, it will add the `promoted-to-master` label, which we need later for the auto backport process
+    # - When ref branch is `branch-x.y` (which means we backported a patch), it will replace in the original PR the `backport/x.y` label with `backport/x.y-done` and will close the backport PR (Since GitHub close only the one referring to default branch)
+    # - Since there is a different with the way we close/merge PR depend on number of commits, this script will handle both scenarios
```

Comment:
could you be more specific? like, what do we expect when handling a PR with a single commit, and what do we expect otherwise. i suspect that it's related to if the query of `https://api.github.com/search/issues` returns no items . but i am not sure if i understand how we are supposed to handle them.

Reply:
This section is no longer relevant